File: docs/guides/ruby/searchRecentlyPublishedBooks.rb
import(time)

require "algolia"

client = Algolia::SearchClient.create("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")

date_timestamp = Time.now.to_i - 60 * 60 * 24 * 365
search_params = Algolia::Search::SearchParamsObject.new(
  query: "<YOUR_SEARCH_QUERY>",
  filters: "date_timestamp > #{date_timestamp}"
)

client.search_single_index("<YOUR_INDEX_NAME>", search_params)

File: docs/guides/dart/lib/searchRecentlyPublishedBooks.dart
import 'package:algolia_client_search/algolia_client_search.dart';

void searchRecentlyPublishedBooks() async {
  final client =
      SearchClient(appId: 'ALGOLIA_APPLICATION_ID', apiKey: 'ALGOLIA_API_KEY');

  final dateTimestamp =
      DateTime.now().subtract(Duration(days: 365)).millisecondsSinceEpoch;
  final searchParams = SearchParamsObject(
      query: "<YOUR_SEARCH_QUERY>", filters: "date_timestamp > $dateTimestamp");

  await client.searchSingleIndex(
    indexName: "<YOUR_INDEX_NAME>",
    searchParams: searchParams,
  );
}
